README.md

```diff
@@ -202,7 +202,101 @@ or for elements of map/array (if it's JSON):
 
 `responseHeaders` - all HTTP response headers for the specified HTTP status codes.
 
+### Variables
 
+You can use variables in the description of the test, they are supported in the following fields:
+
+- method
+- path
+- query
+- headers
+- request
+- response
+
+Example:
+
+```yaml
+- method: "{{ $method }}"
+  path: "/some/path/{{ $pathPart }}"
+  query: "{{ $query }}"
+  headers:
+    header1: "{{ $header }}"
+  request: '{"reqParam": "{{ $reqParam }}"}'
+  response:
+    200: "{{ $resp }}"
+```
+
+You can assign values to variables in the following ways:
+
+- in the description of the test
+- from the response of the previous test 
+- in environment variables or in env-file
+
+Source priorities correspond to the order of enumeration
+
+#### More detailed about assignment methods
+
+##### In the description of the test
+
+Example: 
+
+```yaml
+- method: "{{ $method }}"
+  path: "/some/path/{{ $pathPart }}"
+  variables:
+    reqParam: "reqParam_value"
+    method: "POST"
+    pathPart: "part_of_path"
+    query: "query_val"
+    header: "header_val"
+    resp: "resp_val"
+  query: "{{ $query }}"
+  headers:
+    header1: "{{ $header }}"
+  request: '{"reqParam": "{{ $reqParam }}"}'
+  response:
+    200: "{{ $resp }}"
+```
+
+##### From the response of the previous test 
+
+Example:
+
+```yaml
+# if the response is plain text
+- name: "get_last_post_id"
+  ...
+  variables_to_set:
+          200: "id"
+
+# if the response is JSON
+- name: "get_last_post_info"
+  variables_to_set:
+          200:
+            id: "id"
+            title: "title"
+            authorId: "author_info.id"
+```
+
+Note that if you need to use the value of a nested field, you can specify the path to it:
+> "author_info.id"
+
+You can use any nesting depth.
+
+##### In environment variables or in env-file
+
+Gonkey automatically checks variable exists in the environment variables (in the same case) and takes the value from there, if it exists.
+
+If an env-file is specified, the variables described in it will be added or replace the corresponding environment variables.
+
+Example of an env file (standard syntax):
+```.env
+jwt=some_jwt_value
+secret=my_secret
+password=private_password
+```
+
+env-file is convenient, for example, to extract private information from the test (passwords, keys, etc.)
```

Comment:
env-file can be convenient to hide sensitive information from a test (passwords, keys, etc.)